src/Authenticators/LDAPAuthenticator.php

```diff
@@ -219,4 +219,24 @@ public function getChangePasswordHandler($link)
     {
         return LDAPChangePasswordHandler::create($link, $this);
     }
+
+    public function checkPassword(Member $member, $password, ValidationResult &$result = null)
+    {
+        $result = $result ?: ValidationResult::create();
+
+        /** @var LDAPService $service */
+        $service = Injector::inst()->get(LDAPService::class);
```

Comment:
We could replace that via Setter Dependency Injection to get the server and allow a custom LDAP Service.

Reply:
Good idea, https://github.com/silverstripe/silverstripe-ldap/issues/7